src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -2327,6 +2327,78 @@ public function trustedProxiesRemoteAddr()
             [null, ['REMOTE_ADDR', '2.2.2.2'], ['2.2.2.2']],
         ];
     }
+
+    /**
+     * @dataProvider safePreferenceData
+     */
+    public function testSafePreferred($server, bool $safePreferenceExpected)
+    {
+        $request = new Request([], [], [], [], [], $server);
+
+        $this->assertEquals($safePreferenceExpected, $request->isSafePreferred());
+    }
+
+    public function safePreferenceData()
+    {
+        return [
+            [[], false],
+            [
+                [
+                    'HTTPS' => 'on',
+                ],
+                false,
+            ],
+            [
+                [
+                    'HTTPS' => 'off',
+                    'HTTP_PREFER' => 'safe',
+                ],
+                false,
+            ],
+            [
+                [
+                    'HTTPS' => 'on',
+                    'HTTP_PREFER' => 'safe',
+                ],
+                true,
+            ],
+            [
+                [
+                    'HTTPS' => 'on',
+                    'HTTP_PREFER' => 'unknown-preference',
+                ],
+                false,
+            ],
+            [
+                [
+                    'HTTPS' => 'on',
+                    'HTTP_PREFER' => 'unknown-preference=42, safe',
+                ],
+                true,
+            ],
+            [
+                [
+                    'HTTPS' => 'on',
+                    'HTTP_PREFER' => 'safe, unknown-preference=42',
+                ],
+                true,
+            ],
+            [
+                [
+                    'HTTPS' => 'on',
+                    'HTTP_PREFER' => ['safe'], ['unknown-preference=42'],
```

Comment:
Is it the correct way to mimic a header sent several times (as allowed by the [RFC 7240](https://tools.ietf.org/html/rfc7240#section-2)) ?